velox/common/memory/tests/HashStringAllocatorTest.cpp

```diff
@@ -285,6 +285,37 @@ TEST_F(HashStringAllocatorTest, multipart) {
   allocator_->checkConsistency();
 }
 
+TEST_F(HashStringAllocatorTest, mixedMultipart) {
+  // Create multi-part allocation with a mix of block allocated from Arena and
+  // MemoryPool.
+
+  const std::string shortString(25, 'x');
+  const std::string extraLongString(5'000, 'y');
+
+  ByteStream stream(allocator_.get());
+
+  auto start = allocator_->newWrite(stream);
+  stream.appendStringView(shortString);
+  auto current = allocator_->finishWrite(stream, 0);
+
+  allocator_->extendWrite(current.second, stream);
+
+  ByteRange range;
+  allocator_->newContiguousRange(extraLongString.size(), &range);
+  stream.setRange(range, 0);
+
+  stream.appendStringView(extraLongString);
+  current = allocator_->finishWrite(stream, 0);
+
+  allocator_->extendWrite(current.second, stream);
+  stream.appendStringView(shortString);
```

Comment:
just curious, you mentioned that pool allocations cannot have a continued header (Header::next is not valid). This means in a multipart allocation we can have a header chain with arena allocations at the start and only one pool allocation at the end. What happens if we then want to make a subsequent small allocation that can be accommodated by the arena? would it still allocate it from the pool, or would it fetch it from the arena and append it before the header at the tail end(which points to the pool allocated memory)?

Also, if Header::next is not valid, does that mean we expect it to always be a nullptr or can it be indeterminate? Wondering if we should end the do-while loop if the latter is true.

Reply:
@bikramSingh91 Bikram, thank you for reviewing. Header::next and Header::nextContinued are not the same. Header::next points the block following current block in the Arena. Header::nextContinued points to a block that is a logical continuation of the current block. Blocks allocated from the pool can have continuations (Header::nextContinued) allocated from either Arena or MemoryPool, but they cannot have next or previous Arena blocks.

Reply:
Got it, thanks for the clarification